### DISCUSSION TITLE
Is it possible to setup two annotated SQSListeners one for a regular SQS (AmazonSQSBufferedAsyncClient) & one for a FIFO (AmazonSQSAsync)?

### DISCUSSION THREAD

**Discussion by kevinmcp123** [awspring/spring-cloud-aws#321] [Q&A]
Hello, 

I have a use case whereby I wish to setup two annotated SQSListeners one for a regular SQS (using AmazonSQSBufferedAsyncClient) & one for a FIFO (using AmazonSQSAsync), is this possible?

Reading the documentation it only offers advice to _replace_ the buffered client...

_"To use FIFO queues with Spring Cloud SQS it is recommended to overwrite default SQS client bean with a custom one that is not based on AmazonSQSBufferedAsyncClient"_ - https://docs.awspring.io/spring-cloud-aws/docs/2.4.0/reference/html/index.html#fifo-queue-support
 
 ... but I may wish to configure some buffered properties for the SQS Buffered client?
 
 Many thanks

**Comment by maciejwalkowiak** [comment]
At this stage it is not possible. In 3.0 messaging will be much more flexible, but there won't be a buffered client since it has not been migrated to AWS SDK v2.

If you would like to contribute to 2.4 a change enabling having two clients that would work, that would be a non breaking change, we can consider it.
